source/fundamentals/data-formats/documents.txt

```diff
@@ -14,34 +14,351 @@ Documents
+       .append("name", new BsonString("Gabriel García Márquez"))
+       .append("lastUpdated", new BsonDateTime(new Date(120, 2, 26).getTime()))
+       .append("novels", new BsonArray(Arrays.asList(
+           new BsonDocument().append("title", new BsonString("One Hundred Years of Solitude")).append("yearPublished", new BsonInt32(1967)),
+           new BsonDocument().append("title", new BsonString("Chronicle of a Death Foretold")).append("yearPublished", new BsonInt32(1981)),
+           new BsonDocument().append("title", new BsonString("Love in the Time of Cholera")).append("yearPublished", new BsonInt32(1985))
+      )));
+
+You can insert this document into a collection using the following code:
+
+.. code-block:: java
+   :copyable: true
+
+   // MongoClient mongoClient = ...
+
+   MongoCollection<BsonDocument> collection = database.getCollection("authors", BsonDocument.class);
+   MongoDatabase database = mongoClient.getDatabase("fundamentals_data");
+
+   InsertOneResult result = collection.insertOne(author);
+
+Once you perform a successful insert, you can retrieve the sample document
+data from the collection using the following code:
+
+.. code-block:: java
+   :copyable: true
+
+   import com.mongodb.client.model.Filters;
+   ...
+
+   BsonDocument doc = collection.find(Filters.eq("novels.yearPublished", 1967)).first();
+   if (doc != null) {
+       System.out.println("_id: " + doc.getObjectId("_id").getValue()
+               + ", name: " + doc.getString("name").getValue()
+               + ", lastUpdated: " + new Date(doc.getDateTime("lastUpdated").getValue()));
+
+       doc.getArray("novels").forEach((novel) -> {
+           System.out.println("title: " + novel.asDocument().getString("title").getValue()
+                   + ", yearPublished: " + novel.asDocument().getInt32("yearPublished").getValue());
+       });
+   }
+
+The output should resemble the following:
+
+.. code-block:: none
+   :copyable: false
+
+   _id: 5fb5fad05f734e3794741a35, name: Gabriel García Márquez, lastUpdated: Thu Mar 26 00:00:00 EDT 2020
+   title: One Hundred Years of Solitude, yearPublished: 1967
+   title: Chronicle of a Death Foretold, yearPublished: 1981
+   title: Love in the Time of Cholera, yearPublished: 1985
+
+For more information on retrieving and manipulating MongoDB data, see our
+:doc:`CRUD guide </fundamentals/crud>`.
+
+For more information on this class, see the
+:java-docs:`BsonDocument API documentation </apidocs/bson/org/bson/BsonDocument.html>`.
 
 BasicDBObject
 -------------
 
-- Description
-- Code example of declaring one and appending fields
-- Link to Fundamentals > CRUD for more information
-- Link to API documentation
+The ``BasicDBObject`` class allows you to access and manipulate document data
+using Java types. We recommend that you avoid using this class unless you
+are migrating an application from that uses an older driver version. In
+addition to deprecation, this class has the following limitations:
```

Comment:
Instead of the mildly awkward "in addition to deprecation" phrasing, I wonder if we could:
1. Add (Deprecated) after the section header, akin to the (BETA) label for Realm Sync
2. Mention "deprecated" in the first sentence ("The deprecated ``BasicDBObject`` class") to make the deprecation even more obvious.

We could then just change this sentence to "This class has the following limitations" and, if we really want to hammer in the deprecation idea, add "This class is deprecated" as a new first bullet point in the limitations.

Reply:
Thought it was deprecated, but I guess not.